src/basic/path-lookup.c

```diff
@@ -91,6 +91,41 @@ int xdg_user_data_dir(char **ret, const char *suffix) {
         return 1;
 }
 
+int runtime_directory(char **ret, const char *suffix) {
+        int r;
+        bool apply_suffix = true;
+        _cleanup_free_ char *d = NULL;
+
+        assert(ret);
+        assert(suffix);
+
+        /* Accept $RUNTIME_DIRECTORY as authoritative
+         * If its missing apply the suffix to /run or $XDG_RUNTIME_DIR
+         * depending on whether are root or a regular user.
+         *
+         * Return value indicates whether the suffix was applied or not */
+
+        const char *e = getenv("RUNTIME_DIRECTORY");
+        if (e) {
+                d = strdup(e);
+                apply_suffix = false;
+        } else if (getuid() == 0)
```

Comment:
i don't really like this check. if you run something as root then it could mean "system mode" or it could also mean "per-user mode of user root", hence it sucks as a condition check.

I mean, it might be OK to do a default logic in some cases per default, but if so this should be decided in the main program, not in a library call hidden somewhere like this here.

just make runtime_directory() take a RuntimeScope parameter (which is the generic enum to encode user vs. system scope in our tree).

Then, make sure that RUNTIME_DIRECTORY always takes precedence, and if not set we use check the runtime scope and either use /run/ or the xdg dir.